azkaban-common/src/main/java/azkaban/utils/OsMemoryUtil.java

```diff
@@ -0,0 +1,110 @@
+package azkaban.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Utility class for getting system memory information
+ */
+class OsMemoryUtil {
+  private static final Logger logger = LoggerFactory.getLogger(SystemMemoryInfo.class);
+
+  static final String MEMINFO_FILE = "/proc/meminfo";
```

Comment:
No. It is never designed to work on Mac. The check will be disabled when the file doesn't exist.